dasharo-compatibility/esp-scanning.robot

```diff
@@ -0,0 +1,100 @@
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+ESP001.001 ESP Scan with OS-specific .efi files added
+    [Documentation]    This test aims to verify that any properly added .efi
+    ...    files will have boot menu entries created for them.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ESP001.001 not supported
+
+    Power On
+    Prepare EFI Partition With System Files
+    Power On
+    Enter Boot Menu Tianocore
+    Check Boot Menu For All Supported Systems    normal
+
+ESP002.001 ESP Scan after deleting additional .efi files
+    [Documentation]    This test aims to verify that none of the systems linger
+    ...    on in the boot menu after we've deleted their files from /EFI/.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ESP002.001 not supported
+
+    Power On
+    Clear Out EFI Partition
+    Power On
+    Enter Boot Menu Tianocore
+    Check Boot Menu For All Supported Systems    empty
+
+ESP003.001 ESP Scan ignores OSes on removable media
+    [Documentation]    This test aims to verify that the bootable /EFI
+    ...    partitions of removable media are ignored by the scan and aren't
+    ...    listed in boot menu, except for DTS.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ESP003.001 not supported
+
+    Prepare USB ISO    img_name=CorePlus-current.iso    img_url=${TINYCORE_URL}
+
+    Power On
+    Enter Boot Menu Tianocore
+    ${boot_list}=    Get Boot Menu Construction
+    Should Not Contain Match    ${boot_list}    *CorePlus*
+
+ESP004.001 ESP Scan does not create duplicate entries
+    [Documentation]    This test aims to verify that the firmware will not
+    ...    create duplicate entries, for example, if both shimx64 and grubx64
+    ...    are present for a single OS.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ESP004.001 not supported
+
+    Power On
+    Prepare EFI Partition With System Files
+    Power On
+    Enter Boot Menu Tianocore
+    Check Boot Menu For All Supported Systems    double_entry_check
+
+ESP005.001 ESP Scan detects Dasharo Tools Suite
+    [Documentation]    This test aims to verify that the firmware detects
+    ...    Dasharo Tools Suite boot media and creates a corresponding boot
+    ...    menu entry.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ESP005.001 not supported
+
+    Power On
+    Prepare USB ISO    img_name=dts-base-image-v1.2.8.iso    img_url=${DTS_URL}
+    Power On
+    Enter Boot Menu Tianocore
+    ${boot_list}=    Get Boot Menu Construction
+    Should Contain Match    ${boot_list}    *Dasharo Tools Suite*
+
+
+*** Keywords ***
+Prepare Required Files For Qemu
+    IF    "${MANUFACTURER}" == "QEMU"
+        Run    wget -O ./image.img https://cloud.3mdeb.com/index.php/s/BwLyjHT9fRncXMY/download/image.img
+        Run
+        ...    wget -O ./dts-base-image-v1.2.8.iso https://dl.3mdeb.com/open-source-firmware/DTS/v1.2.8/dts-base-image-v1.2.8.iso
```

Comment:
We have some variables in lib defined, why using hardcoded links here?

Reply:
Corrected in [commit a90d5e0](https://github.com/Dasharo/open-source-firmware-validation/commit/a90d5e01100174c8b883b01ae914d4693f595747#diff-c8f7a5cdc7c64ba736049ef48cf813ae918ab342659418e21d720981c53f903aR94)